auth0/internal/wiremock/client.go

```diff
@@ -3,88 +3,104 @@ package wiremock
 import (
 	"context"
 	"fmt"
+	"log"
 	"net/http"
 	"net/url"
 
 	"github.com/auth0/go-auth0/management"
 	"github.com/hashicorp/go-multierror"
-	"github.com/hashicorp/go-retryablehttp"
 )
 
-// Client provides communciation with the admin API of a WireMock server
-// instance.
+const (
+	resetMappingsURI  = "/__admin/mappings/reset"
+	resetScenariosURI = "/__admin/scenarios/reset"
+)
+
+// Client provides communication with the
+// admin API of a WireMock server instance.
 type Client struct {
-	host string
-	cl   *retryablehttp.Client
+	baseURL    *url.URL
+	httpClient *http.Client
 }
 
-// NewManagementAPIClient creates a Go Auth0 management API client configured to
+// NewClient constructs a WireMock httpClient
+// that communicates with the WireMock
+// server on the given baseURL and port.
+func NewClient(host string) *Client {
+	return &Client{
+		baseURL: &url.URL{
+			Scheme: "http",
+			Host:   host,
+		},
+		httpClient: http.DefaultClient,
+	}
+}
+
+// NewManagementAPIClient creates a Go Auth0 management API httpClient configured to
 // talk to this WireMock server instance.
 func (c *Client) NewManagementAPIClient() (*management.Management, error) {
-	return management.New(c.host, management.WithInsecure())
+	return management.New(c.baseURL.String(), management.WithInsecure())
+}
+
+// Reset calls ResetAllMappings and ResetAllScenarios
+// to provide us with a clean WireMock state.
+func (c *Client) Reset(ctx context.Context) error {
+	return multierror.Append(
+		c.ResetAllMappings(ctx),
+		c.ResetAllScenarios(ctx),
+	).ErrorOrNil()
 }
 
 // ResetAllMappings causes the WireMock server to remove transient mappings and
 // reload static mappings from disk.
-func (c *Client) ResetAllMappings(ctx context.Context) (err error) {
-	resp, err := c.doRequest(ctx, http.MethodPost, "__admin/mappings/reset")
+func (c *Client) ResetAllMappings(ctx context.Context) error {
+	response, err := c.doRequest(ctx, http.MethodPost, resetMappingsURI)
```

Comment:
This is purely a personal preference where whenever possible I prefer to use the entire word for a variable instead of single letters or abbreviations. It reads better IMO.